src/plugins/__tests__/plugins.unit.js

```diff
@@ -0,0 +1,82 @@
+import {
+  getPlugin,
+  getPluginsNames,
+  registerPlugin,
+  registerPluginOnce,
+} from '../plugins';
+
+describe('plugins', () => {
+  describe('registering', () => {
+    it('should register plugin under an alias', () => {
+      const pluginName = 'pluginA';
+      const pluginRef = jest.fn();
+
+      registerPlugin(pluginName, pluginRef);
+      expect(getPlugin(pluginName)).toBe(pluginRef);
+    });
+
+    it('should throw an error if there is already registered plugin at the same name', () => {
+      const pluginName = 'pluginA';
+      const pluginRef = jest.fn();
+
+      expect(() => {
+        registerPlugin(pluginName, pluginRef);
+      }).toThrowError();
+    });
+
+    it('should register plugins in the correct order', () => {
+      const pluginRef = jest.fn();
+
+      registerPlugin('pluginB', pluginRef, 10);
+      registerPlugin('pluginC', pluginRef);
+      registerPlugin('pluginD', pluginRef, 30);
+      registerPlugin('pluginE', pluginRef, 20);
+      registerPlugin('pluginF', pluginRef);
+      registerPlugin('pluginG', pluginRef, 0);
+
+      expect(getPluginsNames()).toEqual([
+        'PluginG',
+        'PluginB',
+        'PluginE',
+        'PluginD',
+        'PluginA',
+        'PluginC',
+        'PluginF',
+      ]);
+    });
```

Comment:
Do you mean this one? https://github.com/handsontable/handsontable/blob/feature/issue-7412/test/e2e/publicAPI.spec.js#L24-L60

Reply:
Close, but no. The linked test checks only if the build has all of them included and registered. Now, we have PLUGIN_PRIORITY and they have to be in the right order. I was thinking about something like the above but for the main Handsontable import where all plugins are loaded and their order is checked:

```ts
expect(getPluginsNames()).toEqual([
...
        'DropdownMenu',
        'Filters',
...
]);
```

Just in case someone will play with PLUGIN_PRIORITY and mix the order. Or if we add a plugin in the future this will force us to think about the place where it should be added. Or if someone forgets to add PLUGIN_PRIORITY this test will fail and remind him that should add the priority.

Reply:
Ok, I got it. At the moment passing tests assures us that plugins are registered in the proper order. I know it's the best explanation but consider the following things. If we add a new plugin - and we skip `PLUGIN_PRIORITY` it will be added at the end (as in usual FIFO) - for most plugins, their order in initialization process doesn't matter. Only if we face a problem with registering hooks we should register them with a priority. Remember that `PLUGIN_PRIORITY` was introduced to provide backwards-compatibility to the plugins we have at the moment. Overall, I'd prefer to aim for priority-less registration (as in editors/renderers/validators/cellTypes).

Reply:
I forgot to mention, we can add:
1) checking `PLUGIN_PRIORITY` for plugins in their test specs,
2) separated unit test file with imported Handsontable from the full package (index.js entry point) and `getPluginsNames()` which one should have access to singletons.
which one do you prefer (or maybe both 🙂)?

Reply:
Yeah, me too. We will work on removing the dependencies. But for now, we have this functionality and should test it (if it makes sense).

> 1. checking `PLUGIN_PRIORITY` for plugins in their test specs,

IMO this won't help. We will only know that it was set, maybe check the value. But that won't tell us if the priority is correct relative to other plugins. I will change the `PLUGIN_PRIORITY` for one plugin and adjust the test. But the order will change and something else may fail. 

> 2. separated unit test file with imported Handsontable from the full package (index.js entry point) and `getPluginsNames()` which one should have access to singletons.

Probably that one? From what you wrote we should check if the first, known items in the array returned from `getPluginsNames()` are correctly ordered and ignore the rest.

But if you think that other tests are enough we can skip this. I'm just afraid that something may slip through. If the order is arbitrary and known then we can just check/enforce it with a test.

Reply:
The second one does its job 💪

Reply:
It's done in https://github.com/handsontable/handsontable/pull/7430/commits/73b1ce608e5211c61d6c92ce4459c36dc03b40a9